src/messages/responses/response-base.ts

```diff
@@ -1,5 +1,51 @@
+import {MomentoErrorCode, SdkError} from '../../errors/errors';
+
 export abstract class ResponseBase {
   public toString(): string {
     return this.constructor.name;
   }
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/ban-types
+type Constructor = new (...args: any[]) => {};
+
+export interface IResponseError {
+  message(): string;
+  innerException(): object;
+  errorCode(): MomentoErrorCode;
+  toString(): string;
+}
+
+export function ResponseError<TBase extends Constructor>(Base: TBase) {
+  return class ResponseError extends Base {
+    public _innerException: SdkError;
+
+    public message(): string {
+      return this._innerException.wrappedErrorMessage();
+    }
+
+    public innerException(): object {
+      return this._innerException;
+    }
+
+    public errorCode(): MomentoErrorCode {
+      return this._innerException.errorCode();
+    }
+
+    public toString(): string {
+      return this.message();
+    }
+  };
+}
+
+export function ResponseHit<TBase extends Constructor>(Base: TBase) {
```

Comment:
I thought I could use them like traits in other languages so functions could require an object to have the `ResponseHit` trait. Doesn't seem to work that way, that's why Error has an interface. There might be ways to make it work more like real traits. I just did it for completeness and in case we do expand the basic implementations or do figure out a way to make them real traits.